### PR TITLE
EWL-6549 Fix category navigation padding and margins

### PR DESCRIPTION
**Pull request by designerneil**
## Ticket(s)

**Jira Ticket**
- [EWL-6549: Category mobile - navigation extending beyond page](https://issues.ama-assn.org/browse/EWL-6549)

## Description
Category MOBILE – Subcategory   Navigation is extending beyond page margins on the left

## To Test
- [ ] switch your branch to `bugfix/EWL-6549-sub-cat-nav`
- [ ] `gulp serve`
- [ ] enable local SG2 testing in your D8
- [ ] `drush @one.local cr`
- [ ] visit http://ama-one.local/advocacy
- [ ] test the category navigation in various viewports and observe the margins and padding
- [ ] observe that the category navigation on mobile is not cut off
- [ ] Did you test in IE 11?

## Visual Regressions

n/a

## Relevant Screenshots/GIFs
screen shot 2018-11-07 at 10 57 00 am

## Remaining Tasks
n/a

## Additional Notes
n/a
---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

File: styleguide/source/assets/scss/03-organisms/_subcategory-exploration.scss [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-  padding-bottom: $gutter / 2;
   overflow: hidden;
+  padding: 0 0 $gutter/4 0;
+
   @include breakpoint($bp-small) {
-    padding-bottom: $gutter;
+    padding: $gutter/2 0;
   }
 
   .ama__subcategory-exploration__list {
```
Comment (nstriedinger) [inline styleguide/source/assets/scss/03-organisms/_subcategory-exploration.scss]:
```suggestion
  &__list {
```

File: styleguide/source/assets/scss/03-organisms/_subcategory-exploration.scss [hunk 2]
```diff
@@ -60,26 +72,22 @@
 .ama__subcategory-exploration__show-more {
```
Comment (nstriedinger) [inline styleguide/source/assets/scss/03-organisms/_subcategory-exploration.scss]:
```suggestion
&__show-more {
```

File: styleguide/source/assets/scss/03-organisms/_subcategory-exploration.scss [hunk 3]
```diff
@@ -60,26 +72,22 @@
...
    @include breakpoint($bp-small) {
-     display: inline-block;
      margin-top: $gutter / 2;
    }
 
-  @include breakpoint($bp-large) {
-    display: none;
-  }
-
   .ama__subcategory-exploration__show-more--close {
```
Comment (nstriedinger) [inline styleguide/source/assets/scss/03-organisms/_subcategory-exploration.scss]:
```suggestion
  &__show-more--close {
```

**Review by nstriedinger — DISMISSED** [review]
@designerneil Can you take a look at these little edits?

File: styleguide/source/assets/js/subcategory-exploration.js [hunk 4]
```diff
@@ -17,8 +17,9 @@
       var $subcategoryListContainerHeight = $subcategoryListContainer.outerHeight();
       var $subcategoryListLinkText = $('.ama__subcategory-exploration__text');
 
-      // If the unordered list height is greater than the parent container then show the show more link
-      if ($subcategoryList.height() > $subcategoryListContainerHeight) {
+      // If the unordered list outerHeight is greater than the parent container then show the show more link
+      if ($subcategoryList.outerHeight() > $subcategoryListContainerHeight) {
+
```
Comment (nstriedinger) [inline styleguide/source/assets/js/subcategory-exploration.js]:
```suggestion
```

**Comment by damontgomery** [comment]
I restarted Travis to see if that helps.

**Review by damontgomery — APPROVED** [review]
This works as described.

I tested on mobile, tablet, and desktop using the Drupal 8 site. Since a lot of the subcategory exploration SCSS was shifted, I reviewed the bottom component as well.

**Review by JaeBea — DISMISSED** [review]
@designerneil ,
There are two things that I believe need updating.
1. The view more text is too low and exposes the very top of hidden subcategory text. See screenshot below.

image

2. The text to the view more link is black and should be purple.

**Comment by designerneil** [comment]
@JaeBea you are right about the link color. I've update that. I can't reproduce the top of the list being exposed. Can you provide steps to repeat?

**Comment by damontgomery** [comment]
https://github.com/AmericanMedicalAssociation/ama-d8/pull/1047 was merged to address the purple color in D8.

**Review by damontgomery — COMMENTED** [review]
This is working mostly as described.

As Beatrice mentioned, there is a very small amount of some of the hidden subcategories showing on mobile.

Example: http://ama-one.local/practice-management

Chrome responsive simulator using Galaxy S5

small_bits

**Comment by designerneil** [comment]
@damontgomery the height on mobile has been addressed. Thanks

**Review by damontgomery — COMMENTED** [review]
This works on mobile now, but on desktop, some of the categories are covered up.

Example: http://ama-one.local/practice-management

medical_practice_management_articles___ama

**Comment by designerneil** [comment]
@damontgomery the height has been resolved. Here is what it looks like now:

Mobile
screen shot 2018-11-15 at 10 45 13 am

Tablet:
screen shot 2018-11-15 at 10 45 26 am

Desktop:
screen shot 2018-11-15 at 10 45 04 am

**Review by damontgomery — APPROVED** [review]
This works as described.

I tested this using the D8 page http://ama-one.local/practice-management on Chrome in Desktop and simulated mobile and on IE11.

**Review by Brian-Clement — APPROVED** [review]
Looking good.  Tested in my local AMA One theme in Chrome, IE using BrowserStack, and ran BackstopJS against production.

**Pull request merged**